File: Programmers/모든_문제/월간코드챌린지/시즌1/20201105/Solution_Level_1_내적.py
def solution(a,b):
    answer = 0
    for i in range(len(a)):
        answer += a[i] * b[i]
    return answer

def solution1(a, b):
    answer = 0
    for x, y in zip(a,b):
        answer += x*y
    return answer
    
def solution2(a, b):    
    return sum([a*b for a, b in zip(a,b)])

if __name__ == "__main__":
    a = [1,2,3,4]
    b = [-3,-1,0,2]
    result = 3
    print(result, solution(a,b), solution1(a,b), solution2(a,b))

    a, b, result = [-1,0,1], [1,0,-1], -2
    print(result, solution(a,b), solution1(a,b), solution2(a,b))
    

File: Programmers/모든_문제/월간코드챌린지/시즌1/20201105/Solution_Level_1_내적.java
public class Solution_Level_1_내적 { // 제출일 2020-11-09 23:02

	public static void main(String[] args) {
		System.out.println(solution(new int[] { 1, 2, 3, 4 }, new int[] { -3, -1, 0, 2 })); // 3
		System.out.println(solution(new int[] { -1, 0, 1 }, new int[] { 1, 0, -1 })); // -2
	}

	// 1000 * 1000 * 1000 = 최대  10억
	public static int solution(int[] a, int[] b) {
		int sum = 0;
		for (int i = 0; i < a.length; i++) {
			sum += a[i] * b[i];
		}
		return sum;
	}

}